sycl/source/detail/program_manager/program_manager.hpp

```diff
@@ -503,7 +506,12 @@ class ProgramManager {
   bool m_UseSpvFile = false;
   RTDeviceBinaryImageUPtr m_SpvFileImage;
 
-  std::set<std::string> m_KernelUsesAssert;
+  // std::less<> is a transparent comparator that enabled comparison between
+  // different types without temporary key_type object creation. This includes
+  // standard overloads, such as comparison between std::string and
+  // std::string_view or just char*.
+  using KernelUsesAssertSet = std::set<std::string, std::less<>>;
```

Comment:
Nit; This alias feels a little overkill when it's just used here and in a unittest.

Reply:
So it is a common practice in C++ to use aliases for container types. As you pointed out, there are already two places. If we decide to change the container type in the future it is easier with alias.